Revert:
2021-03-05  Vladimir N. Makarov  <vmakarov@redhat.com>

	PR target/99378
	* lra-constraints.c (process_address_1): Skip decomposing address
	for asm insn operand with unknown constraint.
    
--- gcc/lra-constraints.c
+++ gcc/lra-constraints.c
@@ -3450,9 +3450,8 @@ process_address_1 (int nop, bool check_only_p,
      i.e. bcst_mem_operand in i386 backend.  */
   else if (MEM_P (mem)
 	   && !(INSN_CODE (curr_insn) < 0
-		&& (cn == CONSTRAINT__UNKNOWN
-		    || (get_constraint_type (cn) == CT_FIXED_FORM
-			&& constraint_satisfied_p (op, cn)))))
+		&& get_constraint_type (cn) == CT_FIXED_FORM
+	        && constraint_satisfied_p (op, cn)))
     decompose_mem_address (&ad, mem);
   else if (GET_CODE (op) == SUBREG
 	   && MEM_P (SUBREG_REG (op)))
